### ISSUE TITLE
Fix the non-deterministic tests

### ISSUE THREAD

**Issue by kelvinst** [henrik/progress_bar#18]
So, there are some tests that randomly fail. I guess that's because they test based on time span, and sometimes the spinner does not render the quantity it was expected to. 

Going deeper: the processor is busy doing something else and the `AnimationServer` depends on it to process the `:tick` message.

Any idea of what could we do to make sure this kind of feature is working? 

Maybe we should not test the timing for spinners at all, only test if it prints at least one step, then any quantity of other steps, and finally the conclusion step.

**Comment by kelvinst** [comment]
It could even make the tests a little faster. 🤣

**Comment by henrik** [comment]
I agree this sucks :) I think I like your idea of checking for some of the steps but maybe not requiring each and every one of them. I'll leave this ticket open and might look at it when things are less busy. Or if anyone wants to make a PR, feel free.
